### DISCUSSION TITLE
understanding question: decentralization or P2P

### DISCUSSION THREAD

**Discussion by tototest99** [markqvist/Reticulum#202] [General]
Hello and many thanks for your inspiring project and your impressive work,

I'm sorry to write such message a bit impatiently, which maybe wrong/lazy due to me not learning/playing enough with the project, but there a quite a few such networking/communication stacks that have pop up recently, so digging deeper into all require a bit too much of time for me...

I have just discovered the project through a recommandation, which stress that Reticulum is decentralized rather than distributed/P2P.
Reading though the [manual section on nodes](https://reticulum.network/manual/understanding.html#node-types), it seems to be indeed the case, isn't it?
If this is correct, I would agree with the following point that if a peer is not required to serve, such networks tend to centralize (control) due to self-interested users being the (vast) majority, as history has empirically shown. On the other hand I understand the argument to leave maximum control to an operator on what data/computation goes on on its machine.

My consequential question would be: 
what do you think about a optional mechanism that allows to check if a peering node is a transport one **and** to only build network with (through after-check, non-acceptance to communicate further with nodes which are not transport ones)?
It would keep the existing feature (allowing to choose to be only an instance node for people who want to) and networks you can build upon,
but add the optional feature to build P2P networks **through coersion** for people who believe in problems aformentionned in the previous paragraph.

**Comment by markqvist** [comment]
Hi @tototest99!

Thanks a lot for the kind words, and for raising this interesting question.

> I have just discovered the project through a recommandation, which stress that Reticulum is decentralized rather than distributed/P2P.

I would say that Reticulum is both decentralised, distributed and peer-to-peer. I absolutely do not subscribe to the idea that you have to *force* everyone to carry out exactly the same work, for something to be a peer-to-peer system.

> if a peer is not required to serve, such networks tend to centralize (control) [...] as history has empirically shown

You are making a pretty big logical fallacy in the conclusions derived from this statement. Yes, we can observe that *some* decentralised or distributed systems have tended towards centralised control. But making the general conclusions that this centralisation of control is a *direct result* of not forcing participants to all do the same work is just plain wrong.

Doing so will **not** fix the problem, it will just exclude 95% of people from participating. You may then *feel* you have solved your "centralisation problem", but in fact you just created a quite fake "equitable system", by excluding everyone else than a small minority (who would have been the holders of centralised power anyway).

(An interesting clue as to why this centralisation *might* occur, which is outside the scope of this discussions, could be found in looking at what all of theses systems were actually *build on top of*: the Internet)

> what do you think about a optional mechanism that allows [...] to build P2P networks through coersion

I think it would only serve to deteriorate and destroy the capacity, functionality, stability and utility of networks built with Reticulum. Forcing everyone to act as transport nodes is, in any network design, not just a bad idea, but quite possibly *the worst* idea. Ever.

You will utterly *destroy* network capacity in *no time flat* by forcing every little wrist-watch, smartphone, grandmas laptop, temperature sensors and other intermittently available devices to transport traffic, with them having absolutely **no idea** of the most efficient way to converge network topography.

*But that is solvable!*, I hear you say, *we can just let them discover all the information they need to know to efficiently transport packets by talking to their peers!*. And how do we do this? By throwing management bandwidth at the problem. *Huge. Amounts. Of. Management. Bandwidth*. What have we just done? Destroyed the network. Would you venture a guess, as to why OLSR, BATMAN-ADV *et al.* never really succeeded?

> [...] for people who believe in problems aformentionned in the previous paragraph

I sincerely hope that people who believe in the importance of these problems will take a good long look at how to *actually solve* them, which is **not by the way of coersion**. That at least, history has, most definitely, "empirically shown".

*I am going to simplify things here, and this stuff really warrants a much more complete exploration and elaboration, but here goes anyway. In simplified form, mind you.*

A much better approach to solving these problems is a combination of the following:

- Make the operation of providing a common resource (network transport, in this example) so cheap, that it is essentially without cost, while at the same time providing benefit to the ones providing it.
- Make sure that there is proper system in place for managing the finite resources being provided (in this case bandwidth). Such a system must:
    1. Have an *understanding of the limits of the resources* within the system
    2. Employ sensible *priorities*, so that when resource limits are close to being reached, those who operate the "resource-providing" system will still be ensured to have access to the resource (again, data transport, in this example).
    3. Allow the free sharing of unused resources with anyone when the system is operating well within limits. Due to the nature of the physical world, *every unused bit of bandwidth is immediately wasted*, so there is **no point in not using it** if it is available. You can't save up data transfer capability for a rainy day, unfortunately (even though some cell carriers would have you believe that).

If those very simple concepts are well-implemented in a system, it can completely regulate itself without any coercion. In the case of a data transport system (a network), it will automatically incentivise people to create more transport nodes, because doing so provides reasonable benefit. When network capacity is getting close to its limits, it makes sense, for those who are capable of it, to install and operate additional transport nodes.

Since there is no extra cost associated with this, and no penalty on their own use, it makes sense to connect and share this bandwidth with everyone else. Both in the short term, and in the long term. Connecting to others from the beginning makes it much more probable that others will alleviate *your* capacity problems later on, and will add new paths for your data to travel through.

Sure, some people will not have the willingness to do this, mostly because *they don't have the ability, knowledge or resources* necessary. But that is ok! In fact, that is how things work best. *Forcing* them is going to solve **nothing**. Trying to create a system that forces *everyone* to do the same thing, is a sort of utterly stupid application of technology-as-blunt-force, that ultimately serves no practical purpose.

Creating a system, where everyone does not have to do the same work, at the same time, all the time, is the basis for *any kind of sustainable society or community*, and our technological systems *must reflect that*.

We are humans, and the border between ourselves and our tools are much more blurry than we commonly perceive. If we build our essential tools and supporting structures on principles of coercion, we become that ourselves, from the individual to the societal level.

One thing that people often misunderstand (in the context of computer and networking systems) is, that while you are *providing* something to other people here, the thing you are providing is being generated *for free* and *completely automatically*, out of an essentially limitless medium. It's not like you have to dig up network transport capacity from the bedrock with your bare hands, or plant it and wait four months for it to sprout, while carefully trimming it's leaves. You set it and forget it. It is so cheap that it is almost preposterous!

If I may venture a bit of a "colorful explanation" here, please allow me: The *potential* to create secure, practically limitless and freely available data transport exists all around you, at all times, and all it takes to make it accessible is to deploy little beacons and lighthouses here and there, that can tap into this *almost magical force*. There is no point in employing some sort of coercive system to access that.
